test/src/graph/nodes/RaytraceNodeTest.cpp

```diff
@@ -21,3 +29,249 @@ TEST_F(RaytraceNodeTest, valid_arguments)
+}
+
+TEST_F(RaytraceNodeTest, config_distortion_invalid_argument_node)
+{
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_distortion(raytraceNode, false), "node != nullptr");
+}
+
+TEST_F(RaytraceNodeTest, config_distortion_invalid_node_object)
+{
+	EXPECT_RGL_INVALID_OBJECT(rgl_node_raytrace_configure_distortion((rgl_node_t) 0x1234, false),
+	                          "Object does not exist: Node 0x1234");
+}
+
+TEST_F(RaytraceNodeTest, config_distortion_valid_arguments)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_distortion(raytraceNode, true));
+}
+
+TEST_F(RaytraceNodeTest, config_non_hit_distance_invalid_argument_node)
+{
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_non_hits(raytraceNode, 0.0f, 0.0f), "node != nullptr");
+}
+
+TEST_F(RaytraceNodeTest, config_non_hit_distance_invalid_node_object)
+{
+	EXPECT_RGL_INVALID_OBJECT(rgl_node_raytrace_configure_non_hits((rgl_node_t) 0x1234, 0.0f, 0.0f),
+	                          "Object does not exist: Node 0x1234");
+}
+
+TEST_F(RaytraceNodeTest, config_non_hit_distance_valid_arguments)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_non_hits(raytraceNode, 0.0f, 0.0f));
+}
+
+TEST_F(RaytraceNodeTest, config_non_hits_should_not_impact_hits)
+{
+	spawnCubeOnScene(Mat3x4f::identity());
+
+	float fovX = 360.0f;
+	float fovY = 180.0f;
+	float resolution = 30.0f;
+	std::vector<rgl_mat3x4f> rays = makeLidar3dRays(fovX, fovY, resolution, resolution);
+
+	std::vector<rgl_field_t> outFields{XYZ_VEC3_F32, IS_HIT_I32, DISTANCE_F32};
+
+	/**
+	 * Hits with no raytrace non hits configuration scenario
+	 */
+
+	rgl_node_t raysNode = nullptr;
+	rgl_node_t yieldNode = nullptr;
+
+	ASSERT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&raysNode, rays.data(), rays.size()));
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+	ASSERT_RGL_SUCCESS(rgl_node_points_yield(&yieldNode, outFields.data(), outFields.size()));
+
+	ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(raysNode, raytraceNode));
+	ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(raytraceNode, yieldNode));
+
+	ASSERT_RGL_SUCCESS(rgl_graph_run(raysNode));
+
+	TestPointCloud expectedPointCloud = TestPointCloud::createFromNode(yieldNode, outFields);
+
+	std::vector<Field<XYZ_VEC3_F32>::type> expectedCoords = expectedPointCloud.getFieldValues<XYZ_VEC3_F32>();
+	std::vector<Field<DISTANCE_F32>::type> expectedDistances = expectedPointCloud.getFieldValues<DISTANCE_F32>();
+	std::vector<Field<IS_HIT_I32>::type> expectedIsHits = expectedPointCloud.getFieldValues<IS_HIT_I32>();
+
+	/**
+	 * Hits with raytrace non hits configuration scenario
+	 */
+
+	rgl_node_t rangeRaysNode = nullptr;
+	std::vector<rgl_vec2f> ranges{
+	    rgl_vec2f{0.0f, CUBE_HALF_EDGE + 1.0f}  // Ray should hit the cube
+	};
+
+	ASSERT_RGL_SUCCESS(rgl_node_rays_set_range(&rangeRaysNode, ranges.data(), ranges.size()));
```

Comment:
Why this additional node is added in the second scenario? I would expect only a change in the non-hits configuration. Now, the test validates not only non-hits configuration but `rgl_node_rays_set_range` also. I would leave the pipeline and the same very the same between runs.